### PR TITLE
fix(docs): search result support scroll to anchor

### PR DESCRIPTION
**Pull request by xyfy**
1. search document in same page
2. search use link to another page(it is some thing like )
3. use same config support http and https

### CODE REVIEW

File: docs/Masa.Docs.Shared/Shared/Toc.razor.cs [hunk 1]
```diff
@@ -79,6 +76,12 @@ private void AppServiceOnTocChanged(object? sender, List<MarkdownItTocContent>?
         _toc = toc.Where(c => c.Level > 1).ToList();
 
         InvokeAsync(StateHasChanged);
+
+        var uri = new Uri(NavigationManager.Uri);
+        if (!string.IsNullOrWhiteSpace(uri.Fragment))
+        {
+            ScrollIntoView(uri.Fragment.Substring(1)).ConfigureAwait(false);
+        }
```
Comment (capdiem) [inline docs/Masa.Docs.Shared/Shared/Toc.razor.cs:84]: 放在NextTick试试，不用等300ms

File: docs/Masa.Docs.Shared/Shared/Toc.razor.cs [hunk 2]
```diff
@@ -30,17 +30,7 @@ protected override async Task OnAfterRenderAsync(bool firstRender)
...
                 await JsRuntime.InvokeVoidAsync("registerWindowScrollEvent", _objRef, ".toc-li");
-                var uri = new Uri(NavigationManager.Uri);
-                if (!string.IsNullOrWhiteSpace(uri.Fragment))
-                {
-                    await NextTickWhile(async () =>
-                    {
-                        await Task.Delay(500);
-                        await ScrollIntoView(uri.Fragment.Substring(1));
-                    }, () => _toc.Count == 0);
-                }
```
Comment (capdiem) [inline docs/Masa.Docs.Shared/Shared/Toc.razor.cs:43]: OnAfterRender的也不能删吧，不然刷新时就不跳锚点了。

**Review by capdiem — CHANGES_REQUESTED** [review]

File: docs/Masa.Docs.Shared/Shared/Toc.razor.cs (hunk 1 above)
Comment (xyfy) [reply docs/Masa.Docs.Shared/Shared/Toc.razor.cs:84]: nexttick效果不稳定,还是delay来得稳定

File: docs/Masa.Docs.Shared/Shared/Toc.razor.cs (hunk 2 above)
Comment (xyfy) [reply docs/Masa.Docs.Shared/Shared/Toc.razor.cs:43]: 会跳

**Review by capdiem — APPROVED** [review]
Good!

**Pull request merged**